frontend/public/src/lib/queries/auth.js

```diff
@@ -19,7 +19,7 @@ export const updateEmailSelector = pathOr(null, ["entities", "updateEmail"])
 const nextState = nthArg(1)
 
 const DEFAULT_OPTIONS = {
-  transform: (auth: AuthResponse) => ({ auth }),
+  transform: (auth: ?AuthResponse) => ({ auth }),
```

Comment:
There was a Flow error:
```
Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈ src/containers/pages/register/RegisterDetailsPage.js:184:5

Cannot call mutateAsync with auth.registerDetailsMutation(...) bound to the
first parameter because AuthResponse [1] is incompatible with null or
undefined [2] in the first argument of property transform.
```
This error is occurring because:
1. Now `redux-query` has flow types...
2. And their flow type [definition for `Transform` is](https://github.com/amplitude/redux-query/blob/562599ee1c482098879ae30e52898862c4639bec/packages/redux-query/src/types.js#L72C8-L72C94)
    ```js
    type Transform = (body: ?ResponseBody, text: ?ResponseText) => { [key: string]: any };
    ```
    which says `body` might be undefined or null.
3. ...whereas we were saying it's not allowed to be.

### Resolution
If `body` ever actually is undefined/null, I suspect we will have issues at runtime. Currently, the transformed object would be `{ auth: undefined }` in that situation. (Maybe that's fine actually. Might just be treated as unauthenticated?) 

Anyway, this doesn't change the runtime behavior, just makes the signature match what redux-query's flow definitions are expecting.

Reply:
Yes, to confirm - that's no logged-in user which is mutated to a not-logged in user, but auth stays undefined because they're not authed